src/day11.rs

```diff
@@ -1,25 +1,375 @@
 // Potential improvements:
 //
 
-pub fn day11(_input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
-    (format!("{}", answer1), format!("{}", answer2))
+use std::{thread, sync::{mpsc, Arc, Mutex}, fmt::Display, collections::BTreeMap};
+
+use rayon::prelude::*;
+use threadpool::ThreadPool;
+
+pub fn day11(input_lines: &[Vec<String>], _solution_ver: &str) -> (String, String) {
+    let grid_sn:i32 = input_lines[0][0].parse().unwrap();
+    let grid:Vec<Vec<i32>> = (1..301).map(|x| {
+        let row = (1..301).map(|y| power_level(x, y, grid_sn)).collect();
+        row
+    }).collect();
+
+    let answer1 = largest_3x3(&grid);
+    let answer2 = largest_nxn(&grid);
+    (format!("{},{}", answer1.0, answer1.1), format!("{},{},{}", answer2.0, answer2.1, answer2.2))
+}
+
+fn power_level(x:i32, y:i32, grid_sn:i32) -> i32{
+    // let out = if (x==90+16 || y==269+16) {true} else {false};
+    let rack_id = x+10;
+    // if out {println!("rack id:{}", rack_id);}
+    let mut pl = rack_id * y;
+    // if out {println!("pl pre sn:{}, sn:{}", pl,grid_sn);}
+    pl += grid_sn;
+    // if out {println!("pl pre mult: {}", pl);}
+    pl *= rack_id;
+    // if out {println!("pl pre 100s char:{}", pl);}
+    pl = hundreds_digit(pl);
+    // if out {println!("pl plus 5:{}", pl);}
+    pl -= 5;
+    // if out {println!("pl:{}", pl);}
+    pl
+}
+
+fn hundreds_digit(pl:i32) -> i32 {
+    pl.to_string().chars().rev().nth(2).unwrap_or_else(|| {println!("Couldn't unwrap, set to 0");'0'}).to_digit(10).unwrap() as i32
+}
+
+fn largest_3x3(grid: &Vec<Vec<i32>>) -> (i32, i32) {
+    let arc_grid = Arc::new(grid);
+    let mut max_power = Mutex::new((0, 0, 0)); // 0th is power, 1st is x, 2nd is y (should go write a struct really..)
+    rayon::scope(|s|
+        for (x_idx, col) in grid.iter().enumerate() {
+            let arc_grid = *arc_grid;
+            let max_power = &max_power;
+            s.spawn(move|s| {
+            for (y_idx, _) in col.iter().enumerate() {
+                if x_idx < 297 && y_idx < 297 {
+                    let mut local_power = 0;
+                    for i in 0..3 {
+                        for j in 0..3 {
+                            local_power += arc_grid[x_idx + i][y_idx + j]
+                        }
+                    }
+                    if local_power > max_power.lock().expect("Failed to get max power mutex for comparison").0 {
+                        *max_power.lock().expect("Failed to get max power mutex to change value") = (local_power, (x_idx+1) as i32, (y_idx+1) as i32);
+                    }
+                }
+                }           
+            });
+        }
+    );
+    let max = *max_power.lock().expect("Failed to get max power mutex to report result");
+
+    (max.1, max.2)
+}
+
+fn largest_nxn(grid: &Vec<Vec<i32>>) -> (i32, i32,i32) {
```

Comment:
It could be useful to provide a comment or two about the overall design approach you've taken.  After reading through it all I can see you've gone the same route as mine (or rather, my second attempt) but I didn't know that until I'd read all the code.  (Obviously this isn't so important for this particular piece of code but given the useful comments you've included elsewhere at a line-by-line level, thought it was worth mentioning for the higher-level feedback!)